3-exceptions/2-after.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+final class PrintFailedException extends \Exception
+{
+}
+
+final class GetReportHandler
+{
+    // ...
+
+    /**
+     * @param int|null $reportId
```

Comment:
null should be removed right

Reply:
The caller still can call with null, but the function checks for null argument and throws exception.

But the more correct way is that we have a query object, and pass that query object to this method, and that query object would have an assert to check the report id is null. But as of this point CQRS is not yet there